packages/nimble-components/src/table/components/cell/index.ts

```diff
@@ -61,6 +77,37 @@ export class TableCell<
         this.menuOpen = event.detail.newState;
         this.$emit('cell-action-menu-toggle', event.detail);
     }
+
+    /**
+     * @internal
+     *
+     * The event handler that is called when a notifier detects a change. Notifiers are added
+     * to the column, so `source` is expected to be an instance of `TableColumn`, and `args`
+     * is the string name of the property that changed on that column.
+     */
+    public handleChange(source: unknown, args: unknown): void {
```

Comment:
Adding observers to every cell just to monitor the columnId feels kind of heavyweight to me.

My vote would be to set column-id for cells in the `TableRow`'s template, similar to how the [table sets record-id on rows in it's template](https://github.com/ni/nimble/blob/bde26d30926b908fcac8f473948b3b360cd6b445/packages/nimble-components/src/table/template.ts#L125).

Reply:
Doing it that way doesn't seem to trigger an update when the column's `columnId` changes. Would you expect it to?

Reply:
Tried it locally and I see it update:
![CellColumnIdChange](https://github.com/ni/nimble/assets/20709258/d16de676-00bd-43c1-851a-5e32fe7e101d)
That's with `@attr({ attribute: 'column-id' }) public columnId?: string;` in `TableCell`, and `column-id="${x => x.column.columnId}"` on the cell in the `row\template.ts`.

Reply:
Oops, I had a failing unit test, but it was because there was no row involved. Was just creating a table cell directly and assigning it a column.